modules/edit-modes/src/lib/rotate-mode.js

```diff
@@ -1,84 +1,195 @@
 // @flow
 
+import turfBuffer from '@turf/buffer';
+import bbox from '@turf/bbox';
 import turfCentroid from '@turf/centroid';
 import turfBearing from '@turf/bearing';
+import bboxPolygon from '@turf/bbox-polygon';
+import turfDistance from '@turf/distance';
+import { coordEach } from '@turf/meta';
+import { getGeom } from '@turf/invariant';
+import { point, featureCollection, lineString } from '@turf/helpers';
 import turfTransformRotate from '@turf/transform-rotate';
 import type {
   PointerMoveEvent,
   StartDraggingEvent,
   StopDraggingEvent,
   DraggingEvent,
-  ModeProps
+  ModeProps,
+  EditHandleFeature
 } from '../types.js';
+import { getPickedEditHandle } from '../utils';
 import type { FeatureCollection, Position } from '../geojson-types.js';
-import { BaseGeoJsonEditMode, type GeoJsonEditAction } from './geojson-edit-mode.js';
+import {
+  BaseGeoJsonEditMode,
+  type GeoJsonEditAction,
+  getIntermediatePosition
+} from './geojson-edit-mode.js';
 import { ImmutableFeatureCollection } from './immutable-feature-collection.js';
 
 export class RotateMode extends BaseGeoJsonEditMode {
-  _isRotatable: boolean;
-  _geometryBeingRotated: ?FeatureCollection;
+  _selectedEditHandle: ?EditHandleFeature;
+  _selectedGeometry: ?FeatureCollection;
+  _previousMouseCoords: ?Position;
+  _isRotating: boolean = false;
+  _isCompositionMode: boolean;
+
+  constructor(options: ?{ isCompositionMode: boolean }) {
```

Comment:
@georgios-uber actually created a CompositeMode. It may need tweaks, but I wonder if TransformMode can use CompositeMode inside itself so we don't need to have these smaller modes aware if they're being composed.